### PR TITLE
Recreate VB tests only based on Features 10 (Review 1st)

### PR DESCRIPTION
**Pull request by paul1956**
Partially Fixes #11655 Replace PR #11819 targeting Features/10.0

## Proposed changes
Add tests to VB Runtime that require no changes to existing code and change existing VB Test to use FluentAssertions 

## Customer Impact
Improves code coverage

## Regression? 

- No

## Risk
- None, only adds test

## Test methodology 
Only adds and improves tests and improves test coverage 

###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11864)

### CODE REVIEW

File: src/Microsoft.VisualBasic.Forms/tests/UnitTests/Microsoft.VisualBasic.Forms.Tests.vbproj [hunk 1]
```diff
@@ -1,14 +1,20 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
-  <PropertyGroup>
+<PropertyGroup>
```
Comment (lonitra) [inline src/Microsoft.VisualBasic.Forms/tests/UnitTests/Microsoft.VisualBasic.Forms.Tests.vbproj:3]: Please revert this change

File: src/Microsoft.VisualBasic.Forms/tests/UnitTests/Microsoft.VisualBasic.Forms.Tests.vbproj [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
     <SourceTargetFramework>$(TargetFramework)</SourceTargetFramework>
     <TargetFramework>$(TargetFramework)-windows7.0</TargetFramework>
     <DisableTransitiveFrameworkReferences>true</DisableTransitiveFrameworkReferences>
-    <RootNamespace>Microsoft.VisualBasic.Forms.Tests</RootNamespace>
+    <RootNamespace></RootNamespace>
```
Comment (lonitra) [inline src/Microsoft.VisualBasic.Forms/tests/UnitTests/Microsoft.VisualBasic.Forms.Tests.vbproj:7]: Let's remove this completely

File: src/Microsoft.VisualBasic.Forms/tests/UnitTests/System/Windows/Forms/TimeTests.vb [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+        <WinFormsFact>
+        Public Sub VbTickCountCloseToEnvironmentTickCount()
+            Dim tickCount As Integer = Math.Abs(Environment.TickCount - My.Computer.Clock.TickCount)
+            Call (tickCount < PrecisionTickLimit).Should.BeTrue()
+        End Sub
+
+        <WinFormsFact>
+        Public Sub VbTimeNotNew()
+            My.Computer.Clock.LocalTime.Should.BeAfter(New Date)
```
Comment (lonitra) [inline src/Microsoft.VisualBasic.Forms/tests/UnitTests/System/Windows/Forms/TimeTests.vb:44]: We don't typically use `My.` in the repo. Correct me if I am wrong, but instead of `My.Computer`, I believe we can just create `Microsoft.VisualBasic.Devices.Computer` to get clock related items. So we can change this test to:
```suggestion
            Dim computer As Computer = New Computer
            computer.Clock.LocalTime.Should.BeAfter(New Date)
```

File: src/Microsoft.VisualBasic.Forms/tests/UnitTests/System/Windows/Forms/TimeTests.vb [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            Public Iterator Function GetEnumerator() As IEnumerator(Of Object()) Implements IEnumerable(Of Object()).GetEnumerator
+                Yield {My.Computer.Clock.GmtTime, Date.UtcNow}
+                Yield {My.Computer.Clock.LocalTime, Date.Now}
+            End Function
+
+            Public Function IEnumerable_GetEnumerator() As IEnumerator Implements IEnumerable.GetEnumerator
+                Return GetEnumerator()
+            End Function
+
+        End Class
```
Comment (lonitra) [inline src/Microsoft.VisualBasic.Forms/tests/UnitTests/System/Windows/Forms/TimeTests.vb:59]: With my comment above, we will not be able to do this anymore so this will need to be removed, and we should inline the needed values into the test

File: src/Microsoft.VisualBasic.Forms/tests/UnitTests/Microsoft.VisualBasic.Forms.Tests.vbproj [hunk 5]
```diff
@@ -1,14 +1,20 @@
...
     <LangVersion>latest</LangVersion>
     <GenerateRuntimeConfigurationFiles>true</GenerateRuntimeConfigurationFiles>
+    <OptionExplicit>On</OptionExplicit>
+    <OptionInfer>Off</OptionInfer>
+    <OptionStrict>On</OptionStrict>
+    <MyType>WindowsFormsWithCustomSubMain</MyType>
+    <UseApplicationFramework>True</UseApplicationFramework>
+    <OutputType>Library</OutputType>
```
Comment (lonitra) [inline src/Microsoft.VisualBasic.Forms/tests/UnitTests/Microsoft.VisualBasic.Forms.Tests.vbproj:15]: These should no longer be needed with my comment above

File: src/Microsoft.VisualBasic.Forms/tests/UnitTests/System/Windows/Forms/TimeTests.vb (hunk 3 above)
Comment (paul1956) [reply src/Microsoft.VisualBasic.Forms/tests/UnitTests/System/Windows/Forms/TimeTests.vb:44]: That is why VB test coverage is low. I am intentionally testing My.

File: src/Microsoft.VisualBasic.Forms/tests/UnitTests/Microsoft.VisualBasic.Forms.Tests.vbproj (hunk 5 above)
Comment (paul1956) [reply src/Microsoft.VisualBasic.Forms/tests/UnitTests/Microsoft.VisualBasic.Forms.Tests.vbproj:15]: That is what I am testing.

File: src/Microsoft.VisualBasic.Forms/tests/UnitTests/System/Windows/Forms/TimeTests.vb (hunk 4 above)
Comment (paul1956) [reply src/Microsoft.VisualBasic.Forms/tests/UnitTests/System/Windows/Forms/TimeTests.vb:59]: I am not testing Date I am testing Application Framework.

File: src/Microsoft.VisualBasic.Forms/tests/UnitTests/System/Windows/Forms/TimeTests.vb (hunk 3 above)
Comment (lonitra) [reply src/Microsoft.VisualBasic.Forms/tests/UnitTests/System/Windows/Forms/TimeTests.vb:44]: Ah, gotcha thanks for the clarification.

File: src/Microsoft.VisualBasic.Forms/tests/UnitTests/Microsoft.VisualBasic.Forms.Tests.vbproj (hunk 1 above)
Comment (paul1956) [reply src/Microsoft.VisualBasic.Forms/tests/UnitTests/Microsoft.VisualBasic.Forms.Tests.vbproj:3]: I think I still need to remove the extra space. Are you resolving because you agree with the rest of the tests?

Comment (lonitra) [reply src/Microsoft.VisualBasic.Forms/tests/UnitTests/Microsoft.VisualBasic.Forms.Tests.vbproj:3]: This is within the project tag so this `<PropertyGroup>` should be indented in once. Other than my remaining comments, changes LGTM 👍

File: src/Microsoft.VisualBasic.Forms/tests/UnitTests/Microsoft.VisualBasic.Forms.Tests.vbproj (hunk 2 above)
Comment (lonitra) [reply src/Microsoft.VisualBasic.Forms/tests/UnitTests/Microsoft.VisualBasic.Forms.Tests.vbproj:7]: I don't think we need to keep an empty `<RootNamespace>` so this is still outstanding.

Comment (paul1956) [reply src/Microsoft.VisualBasic.Forms/tests/UnitTests/Microsoft.VisualBasic.Forms.Tests.vbproj:7]: That came from features/10.0 but I agree it should be removed and I will do that.

Comment (paul1956) [reply src/Microsoft.VisualBasic.Forms/tests/UnitTests/Microsoft.VisualBasic.Forms.Tests.vbproj:7]: VB needs an empty RootNamespace of you get the default Namespace added

**Review by lonitra — APPROVED** [review]
LGTM, thank you!
